src/Tempest/Http/src/Routing/Matching/GenericRouteMatcher.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Http\Routing\Matching;
+
+use Psr\Http\Message\ServerRequestInterface as PsrRequest;
+use Tempest\Http\MatchedRoute;
+use Tempest\Http\Route;
+use Tempest\Http\RouteConfig;
+use Tempest\Http\Routing\Construction\MarkedRoute;
+
+final readonly class GenericRouteMatcher implements RouteMatcher
```

Comment:
This roughly contains everything which the GenericRouter had initially. It translates the incoming request to a `MatchedRoute` including parameters. 

I changed two things:

1. Match using the new regexes and call `prepareMatchingRegexes` to ensure the routing tree is translated (once) to regexes
2. The parameters are extracted from the initial route match instead of matching them seperatly. An old todo in the GenericRouter also suggested this.